flows/interfaces.go

```diff
@@ -206,22 +206,33 @@ type Router interface {
 	Wait() Wait
 	ResultName() string
 
-	PickExit(FlowRun, Step, EventCallback) (ExitUUID, error)
 	Validate([]Exit) error
+	AllowTimeout() bool
+	Route(FlowRun, Step, EventCallback) (ExitUUID, error)
 }
 
 type Exit interface {
 	UUID() ExitUUID
 	DestinationUUID() NodeUUID
 }
 
+type Timeout interface {
+	Seconds() int
+	CategoryUUID() CategoryUUID
+}
+
 type Wait interface {
 	utils.Typed
 
-	Timeout() *int
-	TimeoutOn() *time.Time
+	Timeout() Timeout
+
+	Begin(FlowRun, EventCallback) ActivatedWait
+}
 
-	Begin(FlowRun, EventCallback) bool
+type ActivatedWait interface {
```

Comment:
I think think we can get rid of this long term and callers use events instead:
  * But we need to go update callers like the simulator
  * We still lean on this in the engine to know if caller is trying to resume with a timeout before timeout_on... but I feel like timeout_on should be advisory to the caller, which would then allow things like the simulator having a "timeout" button that the user can click that any time